tests/Backend/Workspaces/MetadataFromSnowflakeWorkspaceTest.php

```diff
@@ -0,0 +1,258 @@
+            'name' => 'languages',
+            'primaryKey' => 'Id',
+        ));
+        // create workspace and source table in workspace
+        $workspaces = new Workspaces($this->_client);
+        $workspace = $workspaces->createWorkspace(["backend" => "snowflake"]);
+        $connection = $workspace['connection'];
+        $db = $this->getDbConnection($connection);
+        $db->query("create table \"test.Languages3\" (
+                \"Id\" integer not null,
+                \"Name\" varchar not null default 'honza',
+                \"update\" varchar
+            );");
+        $db->query("insert into \"test.Languages3\" (\"Id\", \"Name\") values (1, 'cz'), (2, 'en');");
+        $this->_client->writeTableAsyncDirect($table['id'], array(
+            'dataWorkspaceId' => $workspace['id'],
+            'dataTableName' => 'test.Languages3',
+        ));
+        $expected = array(
+            '"Id","Name","update"',
+            '"1","cz",""',
+            '"2","en",""',
+        );
+        $this->assertLinesEqualsSorted(implode("\n", $expected) . "\n", $this->_client->getTableDataPreview($table['id'], array(
+            'format' => 'rfc',
+        )), 'imported data comparsion');
+        // check the created metadata
+        $expectedIdMetadata = [
+            'KBC.datatype.type' => 'NUMBER',
+            'KBC.datatype.nullable' => '',
+            'KBC.datatype.basetype' => 'NUMERIC',
+            'KBC.datatype.length' => '38,0',
+            'KBC.datatype.default' => '',
+        ];
+        $expectedNameMetadata = [
+            'KBC.datatype.type' => 'TEXT',
+            'KBC.datatype.nullable' => '',
+            'KBC.datatype.basetype' => 'STRING',
+            'KBC.datatype.length' => '16777216',
+            'KBC.datatype.default' => '\'honza\'',
+        ];
+        $expectedUpdateMetadata = [
+            'KBC.datatype.type' => 'TEXT',
+            'KBC.datatype.nullable' => '1',
+            'KBC.datatype.basetype' => 'STRING',
+            'KBC.datatype.length' => '16777216',
+            'KBC.datatype.default' => '',
+        ];
+        // check that the new table has the correct metadata
+        $table = $this->_client->getTable($table['id']);
+        $this->assertEquals([], $table['metadata']);
+        $this->assertArrayHasKey('Id', $table['columnMetadata']);
+        $this->assertMetadata($expectedIdMetadata, $table['columnMetadata']['Id']);
+        $this->assertArrayHasKey('Name', $table['columnMetadata']);
+        $this->assertMetadata($expectedNameMetadata, $table['columnMetadata']['Name']);
+        $this->assertArrayHasKey('update', $table['columnMetadata']);
+        $this->assertMetadata($expectedUpdateMetadata, $table['columnMetadata']['update']);
+
+        $db->query("truncate table \"test.Languages3\"");
+        $db->query("alter table \"test.Languages3\" ADD COLUMN \"new_col\" varchar(64)");
+        $db->query("alter table \"test.Languages3\" RENAME TO \"test.Languages4\"");
```

Comment:
What is exact purpose of renaming this table ? :thinking:

Reply:
Ah, it's probably left over from an earlier version where we were also collecting table-level metadata and one of those items was "source table name".  So it doesn't really serve a purpose anymore, but it also doesn't do any harm.

Reply:
ok, in that way I will delete it, because it has no meaning :-)

`$this->assertTrue(true);` will also doesn't do any harm ... if you understand :-)

Reply:
fixed